src/plugins/intel_gpu/src/graph/impls/ocl/adaptive_pooling.cpp

```diff
@@ -0,0 +1,105 @@
+        }
+
+        args.outputs = {instance.output_memory_ptr()};
+        return args;
+    }
+
+public:
+    static primitive_impl* create(const adaptive_pooling_node& arg) {
+        auto params = get_default_params<kernel_selector::adaptive_pooling_params>(arg);
+        auto optional_params = get_default_optional_params<kernel_selector::adaptive_pooling_optional_params>(arg.get_program());
+
+        const auto& primitive = arg.get_primitive();
+        if (primitive->mode == adaptive_pooling_mode::average) {
+            params.mode = kernel_selector::PoolType::AVG;
+        } else {
+            params.mode = kernel_selector::PoolType::MAX;
+
+            switch (primitive->index_element_type) {
+                case cldnn::data_types::i32: {
+                    params.poolIndexElementType = kernel_selector::Datatype::INT32;
+                    break;
+                }
+                case cldnn::data_types::i64: {
+                    params.poolIndexElementType = kernel_selector::Datatype::INT64;
+                    break;
+                }
+                default:
+                    throw std::runtime_error{"Not supported index element type"};
+            }
+
+            params.inputs.push_back(convert_data_tensor(arg.output_indices().get_output_layout()));
+        }
+
+        const auto& kernel_selector = kernel_selector::adaptive_pooling_kernel_selector::Instance();
+        const auto best_kernels = kernel_selector.GetBestKernels(params, optional_params);
+
+        CLDNN_ERROR_BOOL(arg.id(),
+                         "best_kernels.empty()",
+                         best_kernels.empty(),
+                         "Cannot find a proper kernel with this arguments");
+
+        return new adaptive_pooling_impl(arg, best_kernels[0]);
+    }
+};
+
+namespace detail {
+attach_adaptive_pooling_impl::attach_adaptive_pooling_impl() {
+    implementation_map<adaptive_pooling>::add(impl_types::ocl, adaptive_pooling_impl::create, {
+            std::make_tuple(data_types::f16, format::bfyx),
+            std::make_tuple(data_types::f16, format::bfzyx),
+            std::make_tuple(data_types::f16, format::b_fs_yx_fsv16),
+            std::make_tuple(data_types::f16, format::bs_fs_yx_bsv16_fsv16),
+            std::make_tuple(data_types::f16, format::b_fs_zyx_fsv16),
+            std::make_tuple(data_types::f16, format::bs_fs_zyx_bsv32_fsv16),
+            std::make_tuple(data_types::f16, format::b_fs_yx_fsv32),
+            std::make_tuple(data_types::f32, format::bs_fs_yx_bsv32_fsv32),
+
+            std::make_tuple(data_types::f32, format::bfyx),
+            std::make_tuple(data_types::f32, format::bfzyx),
+            std::make_tuple(data_types::f32, format::b_fs_yx_fsv16),
+            std::make_tuple(data_types::f32, format::bs_fs_yx_bsv16_fsv16),
+            std::make_tuple(data_types::f32, format::b_fs_zyx_fsv16),
+            std::make_tuple(data_types::f32, format::bs_fs_zyx_bsv32_fsv16),
```

Comment:
please add this operation to the operation whitelist in program.cpp for can_use_fsv16, can_use_bsv32_fsv...

Reply:
As we discussed today, removed these layouts for now